src/definitions/collections/form.less

```diff
@@ -448,196 +414,214 @@
 ---------------------*/
+each(@formStates, {
+  @state: replace(@key, '@', '');
+  @c: @formStates[@@state][color];
+  @bg: @formStates[@@state][background];
+  @bdc:  @formStates[@@state][borderColor];
+  //@l: @colors[@@color][light];
+  //@h: @colors[@@color][hover];
+  //@lh: @colors[@@color][lightHover];
+
+  /* On Form */
+  .ui.form.@{state} .@{state}.message:not(:empty) {
+    display: block;
+  }
+  .ui.form.@{state} .compact.@{state}.message:not(:empty) {
+    display: inline-block;
+  }
+  .ui.form.@{state} .icon.@{state}.message:not(:empty) {
+    display: flex;
+  }
 
-/* On Form */
-.ui.form.success .success.message:not(:empty) {
-  display: block;
-}
-.ui.form.success .compact.success.message:not(:empty) {
-  display: inline-block;
-}
-.ui.form.success .icon.success.message:not(:empty) {
-  display: flex;
-}
+  /* On Field(s) */
+  .ui.ui.form .fields.@{state} .field label,
+  .ui.ui.form .field.@{state} label,
+  .ui.ui.form .fields.@{state} .field .input,
+  .ui.ui.form .field.@{state} .input {
+    color: @c;
+  }
 
-/*--------------------
-        Warning
----------------------*/
+  .ui.form .fields.@{state} .field .corner.label,
+  .ui.form .field.@{state} .corner.label {
+    border-color: @c;
+    color: @formStates[@@state][cornerLabelColor];
+  }
 
-/* On Form */
-.ui.form.warning .warning.message:not(:empty) {
-  display: block;
-}
-.ui.form.warning .compact.warning.message:not(:empty) {
-  display: inline-block;
-}
-.ui.form.warning .icon.warning.message:not(:empty) {
-  display: flex;
-}
+  .ui.form .fields.@{state} .field textarea,
+  .ui.form .fields.@{state} .field select,
+  .ui.form .fields.@{state} .field input:not([type]),
+  .ui.form .fields.@{state} .field input[type="date"],
+  .ui.form .fields.@{state} .field input[type="datetime-local"],
+  .ui.form .fields.@{state} .field input[type="email"],
+  .ui.form .fields.@{state} .field input[type="number"],
+  .ui.form .fields.@{state} .field input[type="password"],
+  .ui.form .fields.@{state} .field input[type="search"],
+  .ui.form .fields.@{state} .field input[type="tel"],
+  .ui.form .fields.@{state} .field input[type="time"],
+  .ui.form .fields.@{state} .field input[type="text"],
+  .ui.form .fields.@{state} .field input[type="file"],
+  .ui.form .fields.@{state} .field input[type="url"],
+  .ui.form .field.@{state} textarea,
+  .ui.form .field.@{state} select,
+  .ui.form .field.@{state} input:not([type]),
+  .ui.form .field.@{state} input[type="date"],
+  .ui.form .field.@{state} input[type="datetime-local"],
+  .ui.form .field.@{state} input[type="email"],
+  .ui.form .field.@{state} input[type="number"],
+  .ui.form .field.@{state} input[type="password"],
+  .ui.form .field.@{state} input[type="search"],
+  .ui.form .field.@{state} input[type="tel"],
+  .ui.form .field.@{state} input[type="time"],
+  .ui.form .field.@{state} input[type="text"],
+  .ui.form .field.@{state} input[type="file"],
+  .ui.form .field.@{state} input[type="url"] {
```

Comment:
:thinking: i wonder if we could reduce all the the specific input selectors everywhere to

```css
.ui.form .field.@{state} input:not([type="checkbox"]):not(type="radio"),
```

?

Reply:
I would like to keep this PR focusing on adding state variations, to avoid regressions.
Opened up an issue for refactoring them.

I think `hidden`, `button`, `file` and `submit` should be excluded using `:not` too.
However, I am afraid that performance impact of many `:not` selector, especially on low-perf devices.
https://www.sitepoint.com/optimizing-css-id-selectors-and-other-myths/
https://dev.to/gmartigny/comment/8h17